ci/build_docs.sh

```diff
@@ -34,7 +36,7 @@ rapids-mamba-retry install \
   --channel legate \
   --channel legate/label/experimental \
   --channel conda-forge \
-  "legate-boost=*=*_cpu*"
+  "legate-boost=${LEGATEBOOST_VERSION}=*_cpu*"
```

Comment:
CI is choking on the version:
```
Looking for: ['legate-boost==24.09.00[build=*_cpu*]']
<snip>
The following package could not be installed
└─ legate-boost 24.09.00 *_cpu* does not exist (perhaps a typo or a missing channel).
```
But I don't really see why this is different from the tests.  I guess we don't need the `=*_cpu*`, but does it make it not find dev versions?

Reply:
> I guess we don't need the =*_cpu*, but does it make it not find dev versions?

I think you're right! That `==` makes me thing that maybe conda is looking for **exactly** 24.09.00 (no dev versions or other modifiers).

I think that removing it might allow it to float.

I've also put us into a weird position for testing here by pushing tags to the repo like `v24.08.00` that result in new packages having lower versions than older packages (v24.09.00.dev{n}), which is not something that's likely to happen during normal development. I was just trying to be careful in case we weren't able to delete packages from the `legate` channel, but I'm afraid I might have just made this more complicated than it needed to be 😬

Reply:
I just pushed the following changes:

* remove `*_cpu*` tag in this script
* run `rapids-generate-version` first, to update the `VERSION` file from git tags

Those two together should hopefully fix this.

And once we get this all working, I can put up similar changes in `legate-dataframe`.

Reply:
I just deleted those `v24.08.00` / `v24.08.01` tags and re-triggered CI here. I think the changes in this PR will be enough to get this working on pushes to `main` (which I'll test after we merge this).